modules/web/src/app/shared/directives/value-cahnged.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Directive, ElementRef, OnInit} from '@angular/core';
+import {NgControl} from '@angular/forms';
+import {DialogModeService} from '@app/core/services/dialog-mode';
+
+@Directive({
+  selector: '[valueChanged]',
+})
+export class VlaueChangedDirective implements OnInit {
+  private _previousValue: any;
+
+  constructor(private _el: ElementRef, private _control: NgControl, private _isEditDialog: DialogModeService) {}
+
+  ngOnInit(): void {
+    if (!this._isEditDialog.isEditDialog) {
+      return;
+    }
+
+    const element = this._el.nativeElement;
+    this._previousValue = this._control.control.value;
+
+    switch (element.classList[0]) {
```

Comment:
`classList` order can change so instead of using `switch` and `[0]` we should use `if` conditions and use `classList.includes` method.